docs/source/plugin/anatomy.rst

```diff
@@ -0,0 +1,345 @@
+
+There are 3 type of rate limiting:
+
+* per-user: how often a rule triggers for each user
+* per-channel: how often a rule triggers for a given channel
+* globally: how often a rule triggers accross the whole network
+
+Example::
+
+   from sopel import plugin
+
+   @plugin.rule(r'Ah[!?.]?')
+   @plugin.rate(user=2)
+   def you_said_ah(bot, trigger):
+      bot.reply('Ha AH!')
+
+Rule label
+----------
+
+A rule has a label: it will be used for logging, documentation, and internal
+manipulation. There are two cases to consider:
+
+* :term:`Generic rules <Generic rule>` and :term:`URL callbacks <URL callback>`
+  use their :term:`callable <Plugin callable>`'s name by default (i.e. the
+  function's ``__name__``). This can be overriden with the
+  :func:`sopel.plugin.label` decorator.
+* A :term:`Named rule` is already named (by definition), so it uses its name
+  directly as rule label. This can't be overriden by a decorator.
+
+This label is particulary useful for bot owners who want to disable a rule in
+a specific channel. In the following example, the ``say_hello`` rule from the
+``hello`` plugin is disable in the ``#rude`` channel:
+
+.. code-block:: ini
+
+   [#rude]
+   disable_commands = {'hello': ['say_hello']}
+
+The rule in question is defined by the ``hello`` plugin like that::
+
+   @plugin.rule(r'hello!?', r'hi!?', r'hey!?')
+   @plugin.label('say_hello')
+   def handler_hello(bot, trigger):
+      bot.reply('Ha AH!')
+
+
+Plugin callables
+================
+
+When a message from the IRC server matches a :term:`Rule`, Sopel will execute
+its attached :term:`callable <Plugin callable>`. All plugin callables follow
+the same interface:
+
+.. py:function:: plugin_callable(bot, trigger)
+
+   :param bot: the bot's instance
+   :type bot: :class:`sopel.bot.SopelWrapper`
+   :param trigger: the object that triggered the call
+   :type trigger: :class:`sopel.trigger.Trigger`
+
+A callable must accept two positional arguments: a
+:class:`bot <sopel.bot.SopelWrapper>` object, and a
+:class:`trigger <sopel.trigger.Trigger>` object. Both are objects tied to the
+message that matches the rule.
+
+The ``bot`` provides the ability to send messages to the network (to say
+something or to send a specific command such as ``join``), and to check the
+state of the bot such as its settings, memory, or database. It is a context
+aware wrapper around the :class:`~sopel.bot.Sopel` instance.
+
+The ``trigger`` provides information about the line which triggered the rule
+and this callable to be executed.
+
+The return value of a callable is ignored  unless it is
+:const:`sopel.plugin.NOLIMIT`, in which case
+:term:`rate limiting <Rate limiting>` will not be applied for that call.
+(see :func:`sopel.plugin.rate`)
+
+.. note::
+
+   Note that the name can, and should, be anything, and it doesn't have to be
+   called ``plugin_callable``. At least, it should not be called ``callable``
+   as it is a :func:`Python built-in function <callable>`::
```

Comment:
```suggestion
   called ``plugin_callable``. At least, it should not be called ``callable``,
   since that is a :func:`Python built-in function <callable>`::
```